libsolidity/codegen/ABIFunctions.h

```diff
@@ -233,6 +242,19 @@ class ABIFunctions
 	/// Part of @a abiDecodingFunction for array types.
 	std::string abiDecodingFunctionFunctionType(FunctionType const& _type, bool _fromMemory, bool _forUseOnStack);
 
+	/// @returns a function that reads a value type from storage.
+	/// Performs bit mask/sign extend cleanup but not validation.
+	/// @param _splitFunctionTypes if false, returns the address and function signature in a
+	/// single variable.
+	std::string readFromStorage(Type const& _type, size_t _offset, bool _splitFunctionTypes);
+
+	/// @returns a function that extracts a value type from storage slot that has been
+	/// retrieved already.
+	/// Performs bit mask/sign extend cleanup but not validation.
```

Comment:
Should mention this "right aligns" the data.

Reply:
Depends on the data, could also left align.